tests/test_core.py

```diff
@@ -145,10 +145,9 @@ def test_default_diagnostic_filename_is_generated_if_no_name_specified(simple_si
 def test_default_diagnostic_filename_increments_for_multiple_diagnostics(simple_sim):
     """Test read_diagnostic_from_input method in Simulation class"""
     simple_sim.read_diagnostics_from_input()
-    assert simple_sim.diagnostics[0].input_data["directory"] == Path("default_output/")
-    assert simple_sim.diagnostics[0].input_data["filename"] == str(Path("default_output/")
+    assert simple_sim.diagnostics[0].input_data["directory"] == Path("default_output")
+    assert simple_sim.diagnostics[0].input_data["filename"] == str(Path("default_output")
                                                                    / Path("clock0.out"))
-    print(simple_sim.diagnostics)
     input_data = simple_sim.diagnostics[2].input_data
     assert input_data["directory"] == Path("default_output")
```

Comment:
This makes me think we should change the code so that `str(Path("default_output"))` is saved in `input_data["directory"]`, rather than the Path object. I think it's ok for now though...